### PR TITLE
Added update recommendation method

### PR DESCRIPTION
**Pull request by zjw9635468**
Added the update recommendation route
Added the test for the update recommendation route
Changed test_method_not_allowed()

### CODE REVIEW

File: service/routes.py [hunk 1]
```diff
@@ -141,6 +141,34 @@ def delete_recommendations(recommendation_id):
...
 
+######################################################################
+# UPDATE AN EXISTING PET
+######################################################################
+@app.route("/recommendations/<int:recommendation_id>", methods=["PUT"])
+def update_recommendations(recommendation_id):
+    """
+    Update a Recommendation
+
+    This endpoint will update a Recommendation based the body that is posted
```
Comment (Ruochen1105) [inline service/routes.py:152]: typo, and "based on the posted body" would be clearer

File: tests/test_routes.py [hunk 2]
```diff
@@ -162,7 +180,7 @@ def test_not_found(self):
 
     def test_method_not_allowed(self):
         """Test if using an unsupported HTTP method returns a 405 Method Not Allowed"""
-        response = self.client.put("/recommendations/1")
+        response = self.client.post("/recommendations/1")
```
Comment (Ruochen1105) [inline tests/test_routes.py:183]: Maybe a better idea to also test "update without an id," reference: https://github.com/nyu-devops/lab-flask-tdd/blob/91420f2498faa8461ba2f63796f2a7b13aef1071/tests/test_routes.py#L240-L243

File: service/routes.py [hunk 3]
```diff
@@ -141,6 +141,34 @@ def delete_recommendations(recommendation_id):
     return "", status.HTTP_204_NO_CONTENT
 
 
+######################################################################
+# UPDATE AN EXISTING PET
```
Comment (Ruochen1105) [inline service/routes.py:145]: typo

**Comment by Ruochen1105** [comment]
Also, don't forget to connect GitHub PRs to ZenHub issues next time.
image
image

File: tests/test_routes.py (hunk 2 above)
Comment (zjw9635468) [reply tests/test_routes.py]: Agree. I have changed it to post method without an id.

File: service/routes.py (hunk 3 above)
Comment (zjw9635468) [reply service/routes.py]: Fixed.

File: service/routes.py (hunk 1 above)
Comment (Ruochen1105) [reply service/routes.py:152]: based *on*

Comment (zjw9635468) [reply service/routes.py]: Fixed.

**Pull request merged**